src/transactions/AllowTrustOpFrame.cpp

```diff
@@ -99,12 +100,13 @@ AllowTrustOpFrame::setFlagValue(AbstractLedgerTxn& ltx, LedgerKey const& key,
 
 bool
 AllowTrustOpFrame::isAuthRevocationValid(AbstractLedgerTxn& ltx,
-                                         bool& authRevocable)
+                                         bool& authRevocable,
+                                         TransactionResultPayload& resPayload)
 {
     // Load the source account
     LedgerTxn ltxSource(ltx); // ltxSource will be rolled back
     auto header = ltxSource.loadHeader();
-    auto sourceAccountEntry = loadSourceAccount(ltxSource, header);
+    auto sourceAccountEntry = loadSourceAccount(ltxSource, header, resPayload);
```

Comment:
Hmm I'm not sure if I agree. @marta-lokhova might have some more input here, but the upcoming parallel overlay work assumes immutable, lock-less `TransactionFrame` objects. For state queries, each thread will have a BucketList snapshot, where each snapshot is thread local and may not be consistent with other threads' snapshots. Given that we have thread-local state snapshots, global state-dependent caches in a "thread-safe" object seem dangerous. On the other hand, `TransactionResult` is not thread safe and should be local to a single thread or protected by locks, so state caches are appropriate. I think the hash caches are fine because they are based on the immutable Tx frame contents and not BucketList state.

Reply:
I see, I haven't considered that angle when suggesting this. The current implementation is fine then, though I'm still not a fan of shoving all the mutable state into the 'result' object.
Also, do we actually benefit from the cached accounts at all here (especially in the overlay context), given that result is a short-lived object (unlike the whole frame that stays around for a while)?

Reply:
Hmm you're right, calls to `checkValid` are pretty expensive now. I think at apply time the cache doesn't matter since we prefetch the source account anyway, but we probably want to be able to keep the cached account around while a TX is in the queue. I think I'll have to let @marta-lokhova comment more on this though and see what she has in mind wrt to her overlay work, but we probably need to figure this out before merging this.

Reply:
After chatting with @marta-lokhova and looking through the code a bit. we actually only use this cache pre-protocol 8. Since it's purely a backwards compatibility thing, performance doesn't matter. Unfortunately we still need to include it because of a cahing bug pre-protocol 8 that affects replay, but so long as we can pass the parallel catchup mission this should be fine as is.

Reply:
Ok, I see, good to know. 
Still, I feel like the code would benefit from decoupling the cache object from the result payload. It's a bit hard for me to evaluate the impact from a quick glance, but it seems like we carry the resPayload around only for the sake of using `loadAccount`. That's not very good for reading and understanding the code - the interface suggests that the function may modify the result, while the implementation only cares about cache, and only due to a bug. I think even specifically calling out `cachedAccountPreProtocol8` would already be beneficial for the readers, but we may also be able to get rid of `resPayload` at least in a few places (and maybe everywhere where it's only used for `loadAccount`).
I realize that I might be a bit nitpicky here, but the previous tx frame implementation was pretty messy to read and modify (just for Soroban we've had quite a few bugs due to complexity of result management), so I would really like to take this PR as an opportunity of making things more clear to avoid issues in the future.